### ISSUE TITLE
Using form customization to change content type

### ISSUE THREAD

**Issue by pyrographics** [modxbot/migrate#5233]
pyrographics created Redmine issue ID 5233

Using form customization to change content type of static resource does not take effect unless the page settings tab is clicked on prior to creating the resource.  I imagine it happens with all customization.

**Comment by gadamiak** [comment]
gadamiak submitted:

I've just noticed the same issue yesterday. This concerns content disposition too. In other words, setting default content type and content disposition with FC resource/create for static resources doesn't work.

**Comment by splittingred** [comment]
splittingred submitted:

This works fine for me. Can you tell me what you're putting in your default value?

**Comment by pyrographics** [comment]
pyrographics submitted:

I set up a new content type for JPG images and added that content type as the form customization default value.  Just to note, the manager correctly replaces the normal value (HTML) when you switch to the page settings tab before creating the resource but if you just save the resource without first switching tabs it does not.

**Comment by splittingred** [comment]
splittingred submitted:

What "default value" did you give it? The ID of the new Content Type?

I tried saving without clicking the tab and it worked fine for me. What browser?

**Comment by gadamiak** [comment]
gadamiak submitted:

Shaun McCormick wrote:

> What "default value" did you give it? The ID of the new Content Type?

Exactly, in my case it's been "@8@". For me it doesn't work at all for static resources (haven't tested with "regular" resources), even after switching tabs. The content type stays the same at default HTML. I tested it with a super user and limited user. My limited users don't see the settings tab on resource creation, so they don't have a chance to switch tabs, thus I tested it with the tabs enabled. I'm running latest develop branch.

Tested in:
- Opera/9.80 (X11; Linux x86_64; U; pl) Presto/2.9.168 Version/11.50
- Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/534.30 (KHTML, like Gecko) Ubuntu/11.04 Chromium/12.0.742.124
- Ubuntu 11.04 x86_64

**Comment by pyrographics** [comment]
pyrographics submitted:

That makes a difference.  Setting the default value to the content type ID number functions correctly but setting the default value to the content type Name only works if you switch tabs first.  In my case, I set up a new content type with the name JPG and entered JPG into the default value which didn't function unless I switched tabs.  I understand now the content type form customization was only intended to work with ID numbers but partial functionality with Name caused me to assume something was not working correctly.

I used Mozilla 5.0.1/Macintosh
